File: python/examples/perplexity.py
from paysponge import SpongeWallet

from _env import load_dotenv, require_env, sponge_base_url

load_dotenv()

PERPLEXITY_BASE_URL = "https://pplx.x402.paysponge.com"

wallet = SpongeWallet.connect(
    api_key=require_env("SPONGE_API_KEY"),
    base_url=sponge_base_url(),
)
agent_id = wallet.get_agent().id

response = wallet.x402_fetch(
    url=f"{PERPLEXITY_BASE_URL}/search",
    method="POST",
    headers={"Content-Type": "application/json"},
    body={
        "query": "What is Sponge wallet SDK?",
        "max_results": 3,
    },
    preferred_chain="base",
    agent_id=agent_id,
)

data = response.get("data", response) if isinstance(response, dict) else response
results = data.get("results", []) if isinstance(data, dict) else []

print("Perplexity search results")
for result in results:
    print({"title": result.get("title"), "url": result.get("url")})

File: typescript/src/perplexity.ts
import { SpongeWallet } from "@paysponge/sdk";
import { requireEnv, spongeBaseUrl } from "./env.js";

const PERPLEXITY_BASE_URL = "https://pplx.x402.paysponge.com";

const wallet = await SpongeWallet.connect({
  apiKey: requireEnv("SPONGE_API_KEY"),
  baseUrl: spongeBaseUrl(),
});

const response = await wallet.x402Fetch({
  url: `${PERPLEXITY_BASE_URL}/search`,
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: {
    query: "What is Sponge wallet SDK?",
    max_results: 3,
  },
  preferredChain: "base",
});

const data =
  response && typeof response === "object" && "data" in response
    ? (response as { data: unknown }).data
    : response;
const results =
  data && typeof data === "object" && "results" in data
    ? (data as { results: Array<{ title?: string; url?: string }> }).results
    : [];

console.log("Perplexity search results");
for (const result of results) {
  console.log({ title: result.title, url: result.url });
}
